src/filter_tree/filter_tree.c

```diff
@@ -983,8 +983,8 @@ static bool _FilterTree_Compact_Pred
 		SIValue lhs = AR_EXP_Evaluate(node->pred.lhs, NULL);
 		SIValue rhs = AR_EXP_Evaluate(node->pred.rhs, NULL);
 		// Evalute result.
-		int ret = _applyFilter(&lhs, &rhs, node->pred.op);
-		SIValue v = SI_BoolVal(ret);
+		FT_Result ret = _applyFilter(&lhs, &rhs, node->pred.op);
+		SIValue v = ret == FILTER_NULL ? SI_NullVal() : SI_BoolVal(ret);
```

Comment:
please add a short comment explaining the situation, provide an example when a filter would return `FILTER_NULL`